twine/cli.py

```diff
@@ -31,7 +30,6 @@ def list_dependencies_and_versions() -> List[Tuple[str, str]]:
     return [
         ("pkginfo", _installed.Installed(pkginfo).version),
         ("requests", requests.__version__),
-        ("setuptools", setuptools.__version__),
```

Comment:
We're loosing debugging information here. If we need to debug a problem related to `packaging`'s version, we'll not have it unless we include that here

Reply:
We should actually have `importlib_metadata` in here too now that I think about it

Reply:
SGTM. I'll add those.

Reply:
Fixed in 786c0d8dad67e7aeb3dc371af80466ad37faf766. Also proceeded to unify the implementation in 01538db9162978f38268941485a17119392d9e5b.